---
int8 flags
bool success
